tests/test_tag.py

```diff
@@ -155,3 +156,37 @@ def test_push_tag_failed(mock_repo):
     with patch('git.repo.fun.name_to_object'):
         with pytest.raises(exceptions.PushException):
             repo.tag.push("my_tag", "origin")
+
+
+def test_list_tags(mock_repo):
+    """
+   GIVEN GitRepo is initialized with a path and repo
+   WHEN GitTag.list_tags() is called
+   THEN repo.tags is returned
+   """
+    expected = ['tag1', 'tag2']
+    my_local_mock = mock_repo
+    type(my_local_mock).tags = PropertyMock(return_value=expected)
+    repo = GitRepo(repo=my_local_mock)
+
+    test_tags = GitTag(repo, None).list_tags()
```

Comment:
We should use repo.tag like in the other tests, rather than GitTag(), to stay close to the way the library should actually be used.